src/tools/availablePoliciesTool.ts

```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as path from 'path';
+import { ext } from '../extensionVariables';
+
+export class AvailablePoliciesTool implements vscode.LanguageModelTool<{}> {
+    async prepareInvocation(
+        _options: vscode.LanguageModelToolInvocationPrepareOptions<{}>,
+        _token: vscode.CancellationToken
+    ) {
+        return {
+            invocationMessage: 'Getting available APIM policies'
+        };
+    }
+
+    async invoke(
+        _options: vscode.LanguageModelToolInvocationOptions<{}>,
+        _token: vscode.CancellationToken
+    ): Promise<vscode.LanguageModelToolResult> {
+        try {
+            const policiesPath = ext.context.asAbsolutePath(path.join('resources', 'knowledgeBase', 'policies.json'));
+            const policiesContent = fs.readFileSync(policiesPath, 'utf8');
+            const policies = JSON.parse(policiesContent);
+
+            const policyList = Object.entries(policies)
+                .map(([name, description]) => `${name}: ${description}`)
+                .join('\n');
+
+            return new vscode.LanguageModelToolResult([
+                new vscode.LanguageModelTextPart('Here are all available APIM policies:\n\n' + policyList)
+            ]);
+        } catch (error) {
+            return new vscode.LanguageModelToolResult([
```

Comment:
Per my testing in latest 1.100, for error, it is better to throw error instead of returning error message.
Then user will see a cross icon like below:
![image](https://github.com/user-attachments/assets/50fbf59c-e80f-404a-ada7-42d776609348)

Reply:
Thank you, that's a good improvement. Seems the VS Code stable version haven't got this functionality. It shows success for the tool calling even there's exception.
Does it makes sense to you that I hold on resolving this comment until VS Code stable version is updated? I have created a follow up task for this.